tests/src/sanity.rs

```diff
@@ -0,0 +1,326 @@
+use std::{collections::HashMap, path::Path};
+
+use k8s_openapi::api::{batch::v1::Job, core::v1::Pod};
+use kube::{api::ListParams, Api};
+use reqwest::Method;
+use tokio::{
+    io::{AsyncBufReadExt, AsyncReadExt, BufReader},
+    time::{sleep, timeout, Duration},
+};
+
+use crate::utils::*;
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    #[tokio::test]
+    // starts the node(express.js) api server, sends four different requests, validates data,
+    // stops the server and validates if the agent job and pod are deleted
+    async fn test_complete_node_api() {
+        let client = setup_kube_client().await;
+
+        let service_url = get_service_url(&client, "default").await.unwrap();
+        let pod_name = get_nginx_pod_name(&client, "default").await.unwrap();
+        let command = vec!["node", "node-e2e/app.js"];
+        let env: HashMap<&str, &str> = HashMap::new(); // for adding more environment variables
+        let mut server = start_node_server(&pod_name, command, env);
+
+        let mut stderr_reader = BufReader::new(server.stderr.take().unwrap());
+        let child_stdout = server.stdout.take().unwrap();
+
+        setup_panic_hook();
```

Comment:
Why is this necessary?

Reply:
we cannot panic from inside the thread. even if we panic the test doesnt fail so we need to change how the default panic works